pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/deployment.go

```diff
@@ -49,6 +50,7 @@ func ExtractDeployment(d *appsv1.Deployment) *model.Deployment {
 	deploy.ConditionMessage = extractDeploymentConditionMessage(d.Status.Conditions)
 
 	deploy.ResourceRequirements = ExtractPodTemplateResourceRequirements(d.Spec.Template)
+	deploy.Tags = append(deploy.Tags, transformers.RetrieveUnifiedServiceTags(d.ObjectMeta.Labels)...)
```

Comment:
Alternatives:
- refactoring agent-payload to create a common interface
- make the change here and on the collectorLevel to reduce the duplication of the `env:<>` tag per resource

Reply:
this is a simple change - and I'd be comfortable with this solution for now. But I feel like we want to be be moving towards a common interface; there's a lot of copy-pasted common behavior across the resources.